### DISCUSSION TITLE
Use mocking within `@vitest/web-worker`

### DISCUSSION THREAD

**Discussion by freshollie** [vitest-dev/vitest#1486] [Q&A] [answered]
From what I have tested so far, the `web-worker` plugin runs the worker in another thread (process?), thus all mocking (network or otherwise) is not able to be be applied.

In my case, I am using `nock` to mock the underlying network communication which isn't being used by the web worker.

Any alternatives here?

**Comment by sheremet-va** [comment, marked as answer]
`web-worker` runs in the same process/thread your tests are running in

**Reply by freshollie** [reply]
Oh right. I'm switching from jest, and I'm finding my tests are not using the nocked values for the web worker tests, but I will keep digging. I looked at the source and I couldn't tell, thanks for the confirmation.

**Reply by freshollie** [reply]
FYI, I think half my issues were previously trying to find a work around for this https://github.com/vitest-dev/vitest/pull/1487

With this fix, my nock tests work (I was working around by not using the `addEventListener` which may have been causing other issues in my workers)
